### PR TITLE
Add unit tests

### PR DESCRIPTION
**Pull request by sebrandon1**

### CODE REVIEW

File: certification/formatters/generic_test.go [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+	}
+
+	for _, tc := range testCases {
+
+		// Patch the function if we expect an error
+		if tc.marshalIndentFailure {
+			xmlMarshalIndent = func(v interface{}, prefix, indent string) ([]byte, error) {
+				return nil, errors.New("this is an error")
+			}
+		} else {
```
Comment (bcrochet) [inline certification/formatters/generic_test.go:140]: Nit: Could get rid of the else clause by assigning to xmlMarshalIndent ahead of the if statement. Feel free to ignore.

File: certification/formatters/generic_test.go [hunk 2]
```diff
@@ -0,0 +1,168 @@
+package formatters
```
Comment (bcrochet) [inline certification/formatters/generic_test.go:1]: Any particular reason to not use ginkgo/gomega? The rest of the code base is being tested with that framework.

**Review by bcrochet — COMMENTED** [review]
Small nit and a question. Otherwise, thanks for the contrib!

File: certification/formatters/generic_test.go (hunk 2 above)
Comment (sebrandon1) [reply certification/formatters/generic_test.go:1]: I can re-write in Ginkgo if desired.  Didn't know it was a requirement.

Comment (bcrochet) [reply certification/formatters/generic_test.go:1]: Not a hard requirement. Just curious, really.

Comment (sebrandon1) [reply certification/formatters/generic_test.go:1]: I just don't prefer the Ginkgo syntax but I'm good with it either way.

Comment (komish) [reply certification/formatters/generic_test.go:1]: Yep, I think we probably prefer to keep things consistent, but this doesn't introduce any other frameworks other than gotest, and it's a relatively small package. I could go either way. I'm fine accepting it as is.

**Review by komish — APPROVED** [review]
Thanks @sebrandon1 

/lgtm

**Review by bcrochet — APPROVED** [review]
I'm good as is.
/lgtm

File: certification/formatters/generic_test.go (hunk 2 above)
Comment (sebrandon1) [reply certification/formatters/generic_test.go:1]: Thanks!

**Pull request merged**